### ISSUE TITLE
More options for generating availability points

### ISSUE THREAD

**Issue by sarahcd** [nilanjanchatterjee/Background_point_generator#5]
* Rename setting "Type of method for availability point generation" because it is too similar to "Type of method for background point generation": Maybe "Define area within which to generate points"?

 * Allow the user to choose a buffer (in m) around the bbox rectangle or MCP to include (e.g., extend the boundary by 1 km).

* Allow the user to upload a polygon shapefile to define the area.

* Allow the user to provide their own bounding box.

* FUTURE: Allow the user to use output (e.g., custom MCP) from a previous app (would require changes on MoveApps side).

**Comment by sarahcd** [comment]
discussed today: The most flexible addition for now is to add these new settings:
* setting to allow a user-uploaded shapefile defining the area
* setting to define a buffer outside the chosen area to include

**Comment by smthfrmn** [comment]
@sarahcd question about this, in the case where the user is generating the background points and fitting the RSF at the individual level, what are you imagining the uploaded shapefile looks like? (1) does it contain polygons for each of the individuals home ranges for example? (2) or is it just a single large polygon for the entire population outside of which no points, per individual, will be generated? (1) is much more complex in terms of the code and also requires the user to upload a shape file in which the polygons are labelled with the track id present in the move2 object, which feels a bit complicated for them. But let me know what you were imagining/what people expressed to you.

**Comment by nilanjanchatterjee** [comment]
I also agree with @smthfrmn with the options. Based on my interactions with people who want a specific shapefile for generating available points, they want to mask for known non-habitat areas (such as cliffs, settlements, and water bodies). to ensure no available points are generated in those areas. 
Using a shapefile for individuals would be very challenging as a small mismatch in the trackID would return no output.

**Comment by smthfrmn** [comment]
okay in that case I am going to implement two new settings:

1. A file to upload a custom extent, which can be a bounding box, polygon etc in which all background points will be generated
2. A file to upload a custom mask, which will contain one or more polygons in which NO background points will be generate

How does that sound @sarahcd @nilanjanchatterjee?

**Comment by sarahcd** [comment]
I agree with 1 and 2 in the latest comment, the mask idea is a good addition. I agree polygons per individual is more effort for probably less user satisfaction because it is harder and error prone. Currently it was not specifically asked for by users and doesn't have a use case. With the proposed solution, users could run separate workflows if groups of animals/deployments had different areas, which would also be more transparent.

**Comment by nilanjanchatterjee** [comment]
This sounds great to me. The area to generate and not generate the points should be sufficient in addition to the MCP and bounding box. If users really want to generate for separate individual then the option for separate workflows would work.
